pkg/share/manager/json/json.go

```diff
@@ -360,19 +354,16 @@ func (m *mgr) ListShares(ctx context.Context, filters []*collaboration.Filter) (
 	defer m.Unlock()
 	user := ctxpkg.ContextMustGetUser(ctx)
 	for _, s := range m.model.Shares {
-		if utils.UserEqual(user.Id, s.Owner) || utils.UserEqual(user.Id, s.Creator) {
+		if share.IsCreatedByUser(s, user) {
 			// no filter we return earlier
 			if len(filters) == 0 {
 				ss = append(ss, s)
-			} else {
-				// check filters
-				// TODO(labkode): add the rest of filters.
-				for _, f := range filters {
-					if f.Type == collaboration.Filter_TYPE_RESOURCE_ID {
-						if utils.ResourceIDEqual(s.ResourceId, f.GetResourceId()) {
-							ss = append(ss, s)
-						}
-					}
+				continue
+			}
+			// check filters
+			for _, f := range filters {
+				if share.MatchesFilter(s, f) {
```

Comment:
Here we return the share if even a single filter matches (or relation) whereas in `ListReceivedShares`, it's an `and` relation. Why the difference?

Reply:
Good catch, I rewrote the `ListReceivedShares` part and must have missed the other one. I'll change that.

Reply:
Ok, I think now I got the correct behavior done.
Filters of the same type form a disjuntion, a logical OR. Filters of separate type form a conjunction, a logical AND.
For example: (resource_id=1 OR resource_id=2) AND (grantee_type=USER OR grantee_type=GROUP)